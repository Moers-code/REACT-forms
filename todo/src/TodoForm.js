import React, { useState } from "react";

const TodoForm = ({ addList }) => {
	const [formData, setFormData] = useState({ task: "" });

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value,
		}));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		addList({ ...formData });
		setFormData({ task: "" });
	};
	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor="taskName">Task Name</label>
			<input
				id="taskName"
				name="task"
				type="text"
				value={formData.task}
				placeholder="Task Name"
				onChange={handleChange}
			/>
			<button>Add List</button>
		</form>
	);
};

export default TodoForm;
