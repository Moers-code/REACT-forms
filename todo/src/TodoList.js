import React, { useState } from "react";
import Todo from "./Todo";
import TodoForm from "./TodoForm";
import { v4 as uuidv4 } from "uuid";

const TodoList = () => {
	const [todos, setTodos] = useState([]);
	const addList = (newList) => {
		setTodos((todos) => [...todos, { id: uuidv4(), ...newList }]);
	};
	const remove = (id) => {
		setTodos((todos) => todos.filter((lst) => lst.id !== id));
	};
	return (
		<div>
			<TodoForm addList={addList} />
			<ul>
				{todos.map((todoTask) => (
					<Todo
						remove={remove}
						key={todoTask.id}
						id={todoTask.id}
						task={todoTask.task}
					/>
				))}
			</ul>
		</div>
	);
};

export default TodoList;
