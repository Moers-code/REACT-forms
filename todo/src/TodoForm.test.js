import React from "react";
import {
	fireEvent,
	render,
	cleanup,
	getAllByText,
	getByText,
} from "@testing-library/react";
import TodoForm from "./TodoForm";
import TodoList from "./TodoList";

afterEach(cleanup);

it("it should render without crashing", () => {
	render(<TodoForm />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<TodoForm />);
	expect(asFragment()).toMatchSnapshot();
});

it("can add a new list", async () => {
	const mockAddList = jest.fn();
	const { getByLabelText, getByText } = render(
		<TodoForm addList={mockAddList} />
	);
	const listInput = getByLabelText("Task Name");

	const submitButton = getByText("Add List");

	fireEvent.change(listInput, { target: { value: "Grocery" } });

	fireEvent.click(submitButton);
	expect(mockAddList).toHaveBeenCalled();
});
