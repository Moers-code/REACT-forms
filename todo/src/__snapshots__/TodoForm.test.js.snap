// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="taskName"
    >
      Task Name
    </label>
    <input
      id="taskName"
      name="task"
      placeholder="Task Name"
      type="text"
      value=""
    />
    <button>
      Add List
    </button>
  </form>
</DocumentFragment>
`;
